modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvTableStorageTest.java

```diff
@@ -113,7 +114,40 @@ public abstract class AbstractMvTableStorageTest extends BaseMvStoragesTest {
 
     protected StorageHashIndexDescriptor hashIdx;
 
-    protected final CatalogService catalogService = mock(CatalogService.class);
+    private final CatalogService catalogService = mock(CatalogService.class);
+
+    protected final StorageIndexDescriptorSupplier indexDescriptorSupplier = new StorageIndexDescriptorSupplier() {
+        @Override
+        public @Nullable StorageIndexDescriptor get(int indexId) {
+            int catalogVersion = catalogService.latestCatalogVersion();
+
+            CatalogIndexDescriptor indexDescriptor = catalogService.index(indexId, catalogVersion);
+
+            if (indexDescriptor == null) {
+                return null;
+            }
+
+            CatalogTableDescriptor tableDescriptor = catalogService.table(indexDescriptor.tableId(), catalogVersion);
+
+            assertThat(tableDescriptor, is(notNullValue()));
+
+            return StorageIndexDescriptor.create(tableDescriptor, indexDescriptor);
+        }
+    };
+
+    private class TestRow {
```

Comment:
Meow

Reply:
I see IgniteTuple - I lose all of my temper and need to refactor it immediately